src/bad-code.py

```diff
@@ -578,3 +578,60 @@ def Funnel(p,d1,d2):
     
     #sys.stdout=sys.__stdout__
 """
+def create(path,archiveList,xFilesFactor=None,aggregationMethod=None,sparse=False,useFallocate=False):
+  # Set default params
+  if xFilesFactor is None:
+    xFilesFactor = 0.5
+  if aggregationMethod is None:
+    aggregationMethod = 'average'
+
+  #Validate archive configurations...
+  validateArchiveList(archiveList)
+
+  #Looks good, now we create the file and write the header
+  if os.path.exists(path):
+    raise InvalidConfiguration("File %s already exists!" % path)
+  fh = None
+  try:
+    fh = open(path,'wb')
+    if LOCK:
+      fcntl.flock( fh.fileno(), fcntl.LOCK_EX )
+
+    aggregationType = struct.pack( longFormat, aggregationMethodToType.get(aggregationMethod, 1) )
+    oldest = max([secondsPerPoint * points for secondsPerPoint,points in archiveList])
+    maxRetention = struct.pack( longFormat, oldest )
+    xFilesFactor = struct.pack( floatFormat, float(xFilesFactor) )
+    archiveCount = struct.pack(longFormat, len(archiveList))
+    packedMetadata = aggregationType + maxRetention + xFilesFactor + archiveCount
+    fh.write(packedMetadata)
+    headerSize = metadataSize + (archiveInfoSize * len(archiveList))
+    archiveOffsetPointer = headerSize
+
+    for secondsPerPoint,points in archiveList:
+      archiveInfo = struct.pack(archiveInfoFormat, archiveOffsetPointer, secondsPerPoint, points)
+      fh.write(archiveInfo)
+      archiveOffsetPointer += (points * pointSize)
+
+    #If configured to use fallocate and capable of fallocate use that, else
+    #attempt sparse if configure or zero pre-allocate if sparse isn't configured.

```

Comment:
E265 block comment should start with '# '
E501 line too long (81 > 79 characters)